### PR TITLE
PVC UI code enhancement

### LINKED ISSUE #5734: Wait for project selection in namespace drop-down on OCP console
Issue seen with Run `1649886578`
Also take screenshots on important pages.

### PR DESCRIPTION
**Pull request by am-agrawa**
Signed-off-by: am-agrawa <amagrawa@redhat.com>

fixes #5734

### CODE REVIEW

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 1]
```diff
@@ -263,7 +317,18 @@ def pvc_clone_ui(
...
+            expected_text=f"{project_name}",
+            timeout=5,
+        )
+        if wait_for_project:
+            logger.info(f"Select test project {project_name}")
+            self.do_click(
+                format_locator(self.pvc_loc["test-project-link"], project_name)
+            )
+        else:
+            logger.error(f"Project {project_name} not found on UI")
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/pvc_ui.py:331]: Did you need an exception [if `wait_for_project` is `False`]?

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 2]
```diff
@@ -222,7 +265,18 @@ def delete_pvc_ui(self, pvc_name, project_name):
...
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/pvc_ui.py:279]: Do you need an exception [if wait_for_project is False]?

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 1 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py:331]: Do you want me to raise any specific exception here?

Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py:337]: Done

File: ocs_ci/ocs/ui/pvc_ui.py (hunk 2 above)
Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py:284]: Done

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 3]
```diff
@@ -222,7 +270,19 @@ def delete_pvc_ui(self, pvc_name, project_name):
...
+            raise NoSuchElementException
```
Comment (OdedViner) [inline ocs_ci/ocs/ui/pvc_ui.py:285]: Can you add a log to the exception? like `raise NoSuchElementException(f"Project {project_name} not found on UI")`

**Review by OdedViner — APPROVED** [review]

File: ocs_ci/ocs/ui/pvc_ui.py [hunk 4]
```diff
@@ -139,7 +161,18 @@ def pvc_resize_ui(self, project_name, pvc_name, new_size):
         logger.info(f"Search and Select test project {project_name}")
         self.do_click(self.pvc_loc["pvc_project_selector"])
         self.do_send_keys(self.pvc_loc["search-project"], text=project_name)
-        self.do_click(format_locator(self.pvc_loc["test-project-link"], project_name))
+        wait_for_project = self.wait_until_expected_text_is_found(
```
Comment (keesturam) [inline ocs_ci/ocs/ui/pvc_ui.py:164]: @am-agrawa lines 164 to 175 is repeated multiple times. could you please modularize this?

Comment (am-agrawa) [reply ocs_ci/ocs/ui/pvc_ui.py]: Done

**Review by keesturam — APPROVED** [review]
Thanks for making the requested change, Aman. Looks neat!

**Comment by am-agrawa** [comment]
Verified locally.

**Pull request merged**